src/lib/krb5/os/locate_kdc.c

```diff
@@ -582,7 +789,10 @@ locate_server(krb5_context context, const krb5_data *realm,
 
 #ifdef KRB5_DNS_LOOKUP
     if (list.nservers == 0)
-        ret = dns_locate_server(context, realm, &list, svc, transport);
+        ret = dns_locate_server_uri(context, realm, &list, svc, transport);
```

Comment:
If ret is non-zero here, we should goto done.  Otherwise you will discard the fatal error (unless some servers were added before it happened).